pkg/controller/metrics/metrics.go

```diff
@@ -655,4 +744,40 @@ func (ca *clusterAccumulator) addConditionToMap(cond hivev1.ClusterDeploymentCon
 	ca.conditions[cond][clusterType]++
 }
 
+// readMetricsConfig reads the metrics config from the file pointed to by the MetricsConfigFileEnvVar
+// environment variable.
+func readMetricsConfig() (*metricsconfig.MetricsConfig, error) {
+	path := os.Getenv(constants.MetricsConfigFileEnvVar)
+	config := &metricsconfig.MetricsConfig{}
+	if len(path) == 0 {
+		return config, nil
+	}
+
+	fileBytes, err := ioutil.ReadFile(path)
+	if err != nil || len(fileBytes) == 0 {
+		return config, err
+	}
+	if err := json.Unmarshal(fileBytes, config); err != nil {
+		return config, err
+	}
+
+	return config, nil
+}
+
+// logDurationMetric should be used to log duration metrics
+func logDurationMetric(metric *prometheus.HistogramVec, cd *hivev1.ClusterDeployment, time float64) {
+	poolNS := "unknown"
```

Comment:
Can we just leave this as the empty string for non-pool clusters?
```suggestion
	poolNS := ""
```

Reply:
Can't be empty so made it <none>